tests/core/test_training.py

```diff
@@ -121,33 +112,32 @@ async def test_training_script_with_restart_stories(tmp_path: Path, domain_path:
     assert True
 
 
-def configs_for_random_seed_test() -> List[Text]:
-    # define the configs for the random_seed tests
-    return ["data/test_config/ted_random_seed.yaml"]
-
+async def test_random_seed(tmp_path: Path, monkeypatch: MonkeyPatch, domain_path: Text, stories_path: Text):
+    policies_config = {
+        "policies": [
+            {"name": TEDPolicy.__name__, "random_seed": 42},
+            {"name": RulePolicy.__name__},
+        ]
+    }
 
-@pytest.mark.parametrize("config_file", configs_for_random_seed_test())
-async def test_random_seed(
-    tmp_path: Path, config_file: Text, domain_path: Text, stories_path: Text
-):
-    # set random seed in config file to
-    # generate a reproducible training result
+    policy_train = Mock()
+    monkeypatch.setattr(TED, "fit", policy_train)
```

Comment:
Mhm, if we don't let it `fit` then we can't really test that both prediction are the same, can we?

Reply:
apparently, the predictions are identical if we mock `fit` or if we don't.

this is how the predictions (`probs_1` or `probs_2`) look like in both cases:
```
{'scores': [{'action': 'action_listen', 'score': 0.0}, {'action': 'action_restart', 'score': 0.0}, {'action': 'action_session_start', 'score': 0.0}, {'action': 'action_default_fallback', 'score': 0.3}, {'action': 'action_deactivate_loop', 'score': 0.0}, {'action': 'action_revert_fallback_events', 'score': 0.0}, {'action': 'action_default_ask_affirmation', 'score': 0.0}, {'action': 'action_default_ask_rephrase', 'score': 0.0}, {'action': 'action_two_stage_fallback', 'score': 0.0}, {'action': 'action_back', 'score': 0.0}, {'action': '...', 'score': 0.0}, {'action': 'utter_channel', 'score': 0.0}, {'action': 'utter_default', 'score': 0.0}, {'action': 'utter_goodbye', 'score': 0.0}, {'action': 'utter_greet', 'score': 0.0}, {'action': 'some_form', 'score': 0.0}], 'policy': 'policy_1_RulePolicy', 'confidence': 0.3, 'tracker': {'sender_id': '1', 'slots': {'name': None, 'requested_slot': None, 'session_started_metadata': None}, 'latest_message': {'intent': {}, 'entities': [], 'text': None, 'message_id': None, 'metadata': {}}, 'latest_event_time': 1616511256.315918, 'followup_action': None, 'paused': False, 'events': [{'event': 'action', 'timestamp': 1616511256.3158872, 'name': 'action_session_start', 'policy': None, 'confidence': 1.0, 'action_text': None, 'hide_rule_turn': False}, {'event': 'session_started', 'timestamp': 1616511256.315901}, {'event': 'action', 'timestamp': 1616511256.315918, 'name': 'action_listen', 'policy': None, 'confidence': None, 'action_text': None, 'hide_rule_turn': False}], 'latest_input_channel': None, 'active_loop': {}, 'latest_action': {'action_name': 'action_listen'}, 'latest_action_name': 'action_listen'}}
```

Reply:
oh dear 🤦🏻 probably becaues there is no conversation history or user message